### ISSUE TITLE
Multiple errors at first run - Java 11 support (javafx)

### ISSUE THREAD

**Issue by alaincardenas** [mauriciojost/olympus-photosync#60]
Hello Mauricio,

I loved the project from the moment I started searching for an import tool for the Olympus TG series.
I've tried to run the tool from a couple of Windows 10 laptops but I've been unable to get to run.

This are the errors I get when I run some commands:

Running gui-starter I get this:

```
Error: Unable to initialize main class org.mauritania.photosync.starter.gui.GuiStarter
Caused by: java.lang.NoClassDefFoundError: javafx/event/EventTarget
```

Running olympus-photosync I get this:

```
[13:43:55,282 INFO    Starter$] Version: 1.0.0
[13:43:55,691 INFO    Starter$] Loading file configuration: PhotosyncConfig(CameraClientConfig(http,192.168.0.10,80,/DCIM,wlan.*=.*,(.*),(\d+),(\d+),(\d+),(\d+).*,true,None,None),output,Criteria(None,None,None),false,false,false,false,GuiConfig(true,100))
[13:43:55,692 DEBUG        Os$] Detected OS: Windows (property value 'windows 10')
Exception in thread "main" java.lang.NoClassDefFoundError: javafx/event/EventTarget
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at java.base/java.lang.ClassLoader.defineClass1(Native Method)
        at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
        at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
        at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
        at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
        at org.mauritania.photosync.starter.Starter$.startApp(Starter.scala:44)
        at org.mauritania.photosync.starter.Starter$.main(Starter.scala:23)
        at org.mauritania.photosync.starter.Starter.main(Starter.scala)
Caused by: java.lang.ClassNotFoundException: javafx.event.EventTarget
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:521)
```

I have the JAVA SDK 11.0.6 installed.
What am I missing?
The log file is attached!

Regards,
Alain
[default.log](https://github.com/mauriciojost/olympus-photosync/files/4208883/default.log)

**Comment by mauriciojost** [comment]
Hi Alain, Thanks for filing the issue. It could be a problem with the packaging. Could you try an older version? Like 0.16.1 for instance. I will investigate when I get some free time.

**Comment by mauriciojost** [comment]
As per https://stackoverflow.com/questions/52467561/intellij-cant-recognize-javafx-11-with-openjdk-11/52470141#52470141, it seems javafx is no longer part of the jdk since Java 11. I will rename the issue accordingly. Naturally the workaround is to use Java 10.

**Comment by mauriciojost** [comment]
It seems now these dependencies are needed:
```
<dependencies>
    <dependency>
        <groupId>org.openjfx</groupId>
        <artifactId>javafx-controls</artifactId>
        <version>11</version>
    </dependency>
    <dependency>
        <groupId>org.openjfx</groupId>
        <artifactId>javafx-fxml</artifactId>
        <version>11</version>
    </dependency>
</dependencies>
```

**Comment by alaincardenas** [comment]
Thanks a lot for the fast response. I see you started investigating already!

I'll give Java 10 a try.

**Comment by alaincardenas** [comment]
Hi Mauricio,

Installing Java SDK v10 fixed the errors. I've tested it and it worked like a charm on an old TG-4. You can add it to the list of the supported devices.

One thing I noticed is that the documentation for --from and --to has the date format as 'mm-dd-yyyy' , but I was only able to make it work using 'yyyy-mm-dd'

Thanks again for a great tool!

Regards,
Alain

**Comment by mauriciojost** [comment]
Fixed the dates issue reported in #75 recently

**Comment by mauriciojost** [comment]
This should be fixed in 1.1.0. Please take a look and come back to me if any problem.
Thanks for using this app!

**Comment by mauriciojost** [comment]
https://github.com/mauriciojost/olympus-photosync/releases/tag/v1.1.0

Closing this issue as it is fixed. Again, if any problem please reopen this ticket.
